### DISCUSSION TITLE
Application vs. free standing functions

### DISCUSSION THREAD

**Discussion by flosse** [ivanceras/sauron#59] [Q&A] [answered]
Is there a reason why there is the `Application` trait instead of free standing functions `view` and `update` like it is done in Elm?

**Comment by ivanceras** [comment, marked as answer]
I think it could be done with a separate `view` and `update` function, but I think `Application` trait is simpler and easy to work with. The `view` and `update` will need to have the same `Msg` object that needs to be passed around anyway, combining them together in one trait makes it more intuitive.
Do you have a use-case where the `view` and `update` need to be free standing functions?

**Reply by flosse** [reply]
> Do you have a use-case where the `view` and `update` need to be free standing functions?

I'm used to writing seed apps where I split all the components into separate modules, some of which have their own `Msg` but others just have a `view` without an `update` function. I'll probably port a small app to Sauron at some point and see if there are any problems.
